openadapt/window/_macos.py

```diff
@@ -135,16 +141,58 @@ def dump_state(element, elements=None):
 def deepconvert_objc(object):
     """Convert all contents of an ObjC object to Python primitives."""
     value = object
+    handled = False
     if isinstance(object, AppKit.NSNumber):
         value = int(object)
     elif isinstance(object, AppKit.NSArray) or isinstance(object, list):
         value = [deepconvert_objc(x) for x in object]
     elif isinstance(object, AppKit.NSDictionary) or isinstance(object, dict):
-        value = dict(object)
-        for k, v in value.items():
-            value[k] = deepconvert_objc(v)
+        value = {deepconvert_objc(k): deepconvert_objc(v) for k, v in object.items()}
+    elif isinstance(object, AppKit.NSString) or isinstance(object, str):
+        value = str(object)
+    # handle core-foundation class AXValueRef
+    elif isinstance(object, ApplicationServices.AXValueRef):
+        # convert to dict - note: this object is not iterable
+        rep = repr(object)
```

Comment:
Is there no way to retrieve these values via properties? Can you please show the output of `pprint(dir(object))` here?

Reply:
umm...
```
['CAMLType',
 'CAMLTypeForKey_',
 'CAMLTypeSupportedForKey_',
 'CA_addValue_multipliedBy_',
 'CA_archivingValueForKey_',
 'CA_copyNumericValue_',
 'CA_copyRenderValue',
 'CA_copyRenderValueWithColorspace_',
 'CA_distanceToValue_',
 'CA_interpolateValue_byFraction_',
 'CA_interpolateValues___interpolator_',
 'CA_numericValueCount',
 'CA_prepareRenderValue',
 'CA_roundToIntegerFromValue_',
 'CA_validateValue_forKey_',
 'CKAssignToContainerWithID_',
 'CKDescription',
 'CKDescriptionPropertiesWithPublic_private_shouldExpand_',
 'CKDescriptionRedact_avoidShortDescription_',
 'CKExpandedDescription',
 'CKHashedDescription',
 'CKObjectDescriptionRedact_',
 'CKObjectDescriptionRedact_avoidShortDescription_',
 'CKPropertiesDescription',
 'CKPropertiesDescriptionStringFromProperties_',
 'CKRedactedDescription',
 'CKSingleLineDescription',
 'CKUnredactedDescription',
 'IKImageRepresentationWithType_',
 'NSLifeguard_autorelease',
 'NSRepresentation',
 'NSRepresentation',
 'NS_addTiledLayerDescendent_',
 'NS_observationForKeyPath_options_block_',
 'NS_observationForKeyPaths_options_block_',
 'NS_removeTiledLayerDescendent_',
 'NS_tiledLayerVisibleRect',
 'RBSIsXPCObject',
 'SCNUI_name',
 'SCN_setupDisplayLinkWithQueue_screen_policy_',
 '_CKDescriptionWithExpansion_',
 '__NS_swiftOverlay_automaticallyNotifiesObserversForKey_foundationImplementation_',
 '__NS_swiftOverlay_keyPathsForValuesAffectingValueForKey_foundationImplementation_',
 '___tryRetain_OA',
 '__autorelease_OA',
 '__dealloc_zombie',
 '__doc__',
 '__im_afterDelay_',
 '__im_afterDelay_modes_',
 '__im_getInvocation_',
 '__im_onDetachedThread',
 '__im_onMainThread',
 '__im_onMainThreadIfNecessary',
 '__im_onThread_',
 '__im_onThread_immediateForMatchingThread_',
 '__module__',
 '__msv_inspect',
 '__msv_inspectWithOptions_',
 '__msv_ivars',
 '__msv_methods',
 '__msv_objectDescription',
 '__msv_properties',
 '__pyobjc_PythonObject__',
 '__pyobjc_PythonObject__',
 '__pyobjc_PythonTransient___',
 '__pyobjc_protocols__',
 '__release_OA',
 '__retain_OA',
 '__slots__',
 '_abCompareContainsSubString_options_',
 '_abCompareNotWithinIntervalAroundTodayYearless_',
 '_abCompareNotWithinIntervalAroundToday_',
 '_abCompareNotWithinIntervalFromTodayYearless_',
 '_abCompareNotWithinIntervalFromToday_',
 '_abComparePrefixMatch_options_',
 '_abCompareSuffixMatch_options_',
 '_abCompareWithinIntervalAroundTodayYearless_',
 '_abCompareWithinIntervalAroundToday_',
 '_abCompareWithinIntervalFromTodayYearless_',
 '_abCompareWithinIntervalFromToday_',
 '_abCompare_ascending_options_',
 '_accessibilityActionDelegate',
 '_accessibilityArrayAttributeCount_clientError_',
 '_accessibilityArrayAttributeValues_index_maxCount_clientError_',
 '_accessibilityAttributeNamesClientError_',
 '_accessibilityCanSetValueForAttribute_clientError_',
 '_accessibilityChildUIElementForSpecifierComponent_',
 '_accessibilityCustomDataBlob',
 '_accessibilityIndexOfChild_clientError_',
 '_accessibilityIsTableViewDescendant',
 '_accessibilitySetOverrideCustomActions_',
 '_accessibilitySetOverrideHandler_forAttribute_',
 '_accessibilitySetOverrideIsAccessibilityElement_',
 '_accessibilitySetOverrideValue_forAttribute_',
 '_accessibilitySetUseConvenienceAPI_',
 '_accessibilityShouldBeTreatedAsButtonForToolbar',
 '_accessibilitySpecifierComponentForChildUIElement_registerIfNeeded_',
 '_accessibilityUIElementSpecifier',
 '_accessibilityUIElementSpecifierForChild_registerIfNeeded_',
 '_accessibilityUIElementSpecifierRegisterIfNeeded_',
 '_accessibilityUseConvenienceAPI',
 '_accessibilityValueForAttribute_clientError_',
 '_addObserver_forProperty_options_context_',
 '_addOptionValue_toArray_withKey_type_',
 '_addPlaceholderOptionValue_isDefault_toArray_withKey_binder_binding_',
 '_allowsDirectEncoding',
 '_asScriptTerminologyNameArray',
 '_asScriptTerminologyNameString',
 '_associatedUniversalLink',
 '_bind_toController_withKeyPath_valueTransformerName_options_existingNibConnectors_connectorsToRemove_connectorsToAdd_',
 '_binderClassForBinding_withBinders_',
 '_binderForBinding_withBinders_createAutoreleasedInstanceIfNotFound_',
 '_binderWithClass_withBinders_createAutoreleasedInstanceIfNotFound_',
 '_bindingAdaptor',
 '_bindingInformationWithExistingNibConnectors_availableControllerChoices_',
 '_cfTypeID',
 '_changeValueForKey_key_key_usingBlock_',
 '_changeValueForKey_usingBlock_',
 '_changeValueForKeys_count_maybeOldValuesDict_maybeNewValuesDict_usingBlock_',
 '_cleanupBindingsWithExistingNibConnectors_exception_',
 '_compatibility_takeValue_forKey_',
 '_conformsToProtocolNamed_',
 '_copyDescription',
 '_createKeyValueBindingForKey_name_bindingType_',
 '_destroyObserverList',
 '_didChangeValuesForKeys_',
 '_didEndKeyValueObserving',
 '_implicitObservationInfo',
 '_intents_compareValue_relation_',
 '_intents_displayImageForLanguage_',
 '_intents_displayImageWithLocalizer_',
 '_intents_enumerateObjectsOfClass_withBlock_',
 '_intents_indexingRepresentation',
 '_intents_isValidKey_',
 '_intents_localizedCopyForLanguage_',
 '_intents_localizedCopyWithLocalizer_',
 '_intents_readableDescriptionForLanguage_',
 '_intents_readableDescriptionForLanguage_withMetadata_',
 '_intents_readableDescriptionWithLocalizer_',
 '_intents_readableDescriptionWithLocalizer_metadata_',
 '_intents_readableSubtitleForLanguage_',
 '_intents_readableSubtitleForLanguage_withMetadata_',
 '_intents_readableSubtitleWithLocalizer_',
 '_intents_readableSubtitleWithLocalizer_metadata_',
 '_intents_readableTitleForLanguage_',
 '_intents_readableTitleForLanguage_withMetadata_',
 '_intents_readableTitleWithLocalizer_',
 '_intents_readableTitleWithLocalizer_metadata_',
 '_intents_setterForPropertyWithName_',
 '_invokeSelector_withArguments_onKeyPath_',
 '_isAXConnector',
 '_isAccessibilityCandidateForSection_',
 '_isAccessibilityContainerSectionCandidate',
 '_isAccessibilityContentNavigatorSectionCandidate',
 '_isAccessibilityContentSectionCandidate',
 '_isAccessibilityTopLevelNavigatorSectionCandidate',
 '_isDeallocating',
 '_isDeallocating',
 '_isKVOA',
 '_isToManyChangeInformation',
 '_localClassNameForClass',
 '_notifyObserversForKeyPath_change_',
 '_notifyObserversOfChangeFromValuesForKeys_toValuesForKeys_',
 '_observerStorage',
 '_observerStorageOfSize_',
 '_oldValueForKeyPath_',
 '_oldValueForKey_',
 '_optionDescriptionsForBinding_',
 '_overrideUseFastBlockObservers',
 '_pendingChangeNotificationsArrayForKey_create_',
 '_placeSuggestionsInDictionary_acceptableControllers_boundBinders_binder_binding_',
 '_pyobjc_performOnThreadWithResult_',
 '_pyobjc_performOnThread_',
 '_ql_waitForKey_toBeEqualTo_',
 '_ql_waitForKey_toBeEqualTo_timeout_',
 '_receiveBox_',
 '_releaseBindingAdaptor',
 '_removeObserver_forProperty_',
 '_scriptingAddObjectsFromArray_toValueForKey_',
 '_scriptingAddObjectsFromSet_toValueForKey_',
 '_scriptingAddToReceiversArray_',
 '_scriptingAlternativeValueRankWithDescriptor_',
 '_scriptingArrayOfObjectsForSpecifier_',
 '_scriptingCanAddObjectsToValueForKey_',
 '_scriptingCanHandleCommand_',
 '_scriptingCanInsertBeforeOrReplaceObjectsAtIndexes_inValueForKey_',
 '_scriptingCanSetValue_forSpecifier_',
 '_scriptingCoerceValue_forKey_',
 '_scriptingCopyWithProperties_forValueForKey_ofContainer_',
 '_scriptingCount',
 '_scriptingCountNonrecursively',
 '_scriptingCountOfValueForKey_',
 '_scriptingDebugDescription',
 '_scriptingDescriptorOfComplexType_orReasonWhyNot_',
 '_scriptingDescriptorOfEnumeratorType_orReasonWhyNot_',
 '_scriptingDescriptorOfObjectType_orReasonWhyNot_',
 '_scriptingDescriptorOfValueType_orReasonWhyNot_',
 '_scriptingExists',
 '_scriptingIndexOfObjectForSpecifier_',
 '_scriptingIndexOfObjectWithName_inValueForKey_',
 '_scriptingIndexOfObjectWithUniqueID_inValueForKey_',
 '_scriptingIndexesOfObjectsForSpecifier_',
 '_scriptingIndicesOfObjectsAfterValidatingSpecifier_',
 '_scriptingIndicesOfObjectsForSpecifier_count_',
 '_scriptingInsertObject_inValueForKey_',
 '_scriptingInsertObjects_atIndexes_inValueForKey_',
 '_scriptingMightHandleCommand_',
 '_scriptingObjectAtIndex_inValueForKey_',
 '_scriptingObjectCountInValueForKey_',
 '_scriptingObjectForSpecifier_',
 '_scriptingObjectWithName_inValueForKey_',
 '_scriptingObjectWithUniqueID_inValueForKey_',
 '_scriptingObjectsAtIndexes_inValueForKey_',
 '_scriptingRemoveAllObjectsFromValueForKey_',
 '_scriptingRemoveObjectsAtIndexes_fromValueForKey_',
 '_scriptingRemoveValueForSpecifier_',
 '_scriptingReplaceObjectAtIndex_withObjects_inValueForKey_',
 '_scriptingSetOfObjectsForSpecifier_',
 '_scriptingSetValue_forKey_',
 '_scriptingSetValue_forSpecifier_',
 '_scriptingShouldCheckObjectIndexes',
 '_scriptingValueForKey_',
 '_scriptingValueForSpecifier_',
 '_setBindingAdaptor_',
 '_setObject_forBothSidesOfRelationshipWithKey_',
 '_setterForProperty_',
 '_shouldSearchChildrenForSection',
 '_suggestedControllerKeyForController_binding_',
 '_supportsGetValueWithNameForKey_perhapsByOverridingClass_',
 '_supportsGetValueWithUniqueIDForKey_perhapsByOverridingClass_',
 '_tryRetain',
 '_tryRetain',
 '_unbind_existingNibConnectors_connectorsToRemove_connectorsToAdd_',
 '_viewIdentifier',
 '_web_description',
 '_webkit_invokeOnMainThread',
 '_willBeginKeyValueObserving',
 '_willChangeValuesForKeys_',
 'abCaseInsensitiveIsEqual_',
 'abDictionaryWithValuesForKeyPaths_',
 'abRemoveObserverIgnoringExceptions_forKeyPath_',
 'accessibilityAddTemporaryChild_',
 'accessibilityAllowsOverriddenAttributesWhenIgnored',
 'accessibilityArrayAttributeCount_',
 'accessibilityArrayAttributeValues_index_maxCount_',
 'accessibilityAttributeValue_forParameter_',
 'accessibilityAttributedValueForStringAttributeAttributeForParameter_',
 'accessibilityBrailleMapRenderRegion',
 'accessibilityBrailleMapRenderer',
 'accessibilityDecodeOverriddenAttributes_',
 'accessibilityEncodeOverriddenAttributes_',
 'accessibilityIndexForChildUIElementAttributeForParameter_',
 'accessibilityIndexOfChild_',
 'accessibilityOverriddenAttributes',
 'accessibilityParameterizedAttributeNames',
 'accessibilityPerformShowMenuOfChild_',
 'accessibilityPresenterProcessIdentifier',
 'accessibilityRemoveTemporaryChild_',
 'accessibilityReplaceRange_withText_',
 'accessibilitySetOverrideValue_forAttribute_',
 'accessibilitySetPresenterProcessIdentifier_',
 'accessibilityShouldSendNotification_',
 'accessibilityShouldUseUniqueId',
 'accessibilitySupportsCustomElementData',
 'accessibilitySupportsNotifications',
 'accessibilitySupportsOverriddenAttributes',
 'accessibilityTemporaryChildren',
 'accessibilityVisibleArea',
 'addChainedObservers_',
 'addObject_toBothSidesOfRelationshipWithKey_',
 'addObject_toPropertyWithKey_',
 'addObservationTransformer_',
 'addObserverBlock_',
 'addObserver_',
 'addObserver_forKeyPath_options_context_',
 'addObserver_forObservableKeyPath_',
 'akToolbarButtonItemType',
 'allPropertyKeys',
 'allowsWeakReference',
 'associatedObject',
 'attributeKeys',
 'autoContentAccessingProxy',
 'autorelease',
 'awakeAfterUsingCoder_',
 'awakeFromNib',
 'bind_toObject_withKeyPath_options_',
 'boolValueSafe',
 'boolValueSafe_',
 'bs_isPlistableType',
 'bs_secureEncoded',
 'bundleForClass',
 'canRenderWithCGLContext_',
 'ck_bindInStatement_atIndex_',
 'cksqlcs_appendSQLConstantValueToString_',
 'cksqlcs_archivedObjectBindingValue_',
 'cksqlcs_bindArchivedObject_index_db_',
 'cksqlcs_bindBlob_index_db_',
 'cksqlcs_bindDouble_index_db_',
 'cksqlcs_bindInt64_index_db_',
 'cksqlcs_bindText_index_db_',
 'cksqlcs_blobBindingValue_destructor_error_',
 'cksqlcs_doubleBindingValue_',
 'cksqlcs_int64BindingValue_',
 'cksqlcs_textBindingValue_destructor_error_',
 'classCode',
 'classDescription',
 'classDescriptionForDestinationKey_',
 'classForArchiver',
 'classForCoder',
 'classForKeyedArchiver',
 'classForPortCoder',
 'className',
 'class__',
 'clearProperties',
 'coerceValueForScriptingProperties_',
 'coerceValue_forKey_',
 'compositionParameterView_didChangeParameterWithKey_',
 'compositionParameterView_shouldDisplayParameterWithKey_attributes_',
 'compositionPickerViewDidStartAnimating_',
 'compositionPickerViewWillStopAnimating_',
 'compositionPickerView_didLoadComposition_',
 'compositionPickerView_didSelectComposition_',
 'compositionPickerView_draggingEnteredComposition_sender_',
 'compositionPickerView_keyDown_',
 'compositionPickerView_performDragOperationOnComposition_sender_',
 'compositionPickerView_willSelectComposition_',
 'conformsToProtocol_',
 'copy',
 'copyRenderedTextureForCGLContext_pixelFormat_bounds_isFlipped_',
 'copyScriptingValue_forKey_withProperties_',
 'createImageWithOptions_',
 'createKeyValueBindingForKey_typeMask_',
 'createOptimizedProviderWithTransformation_cropping_',
 'dealloc',
 'dealloc',
 'debugDescription',
 'description',
 'description',
 'description',
 'descriptionAtIndent_',
 'dictionaryWithValuesForKeys_',
 'didChangeValueForKey_',
 'didChangeValueForKey_withSetMutation_usingObjects_',
 'didChange_valuesAtIndexes_forKey_',
 'doesContain_',
 'doesNotRecognizeSelector_',
 'doesNotRecognizeSelector_',
 'doubleValueSafe',
 'doubleValueSafe_',
 'encodeWithCAMLWriter_',
 'entityName',
 'exposedBindings',
 'finalize',
 'finishObserving',
 'flushKeyBindings',
 'forwardInvocation_',
 'forwardingTargetForSelector_',
 'fp__ivarDescriptionForClass_',
 'fp__methodDescriptionForClass_',
 'fp_ivarDescription',
 'fp_methodDescription',
 'fp_shortMethodDescription',
 'handleQueryWithUnboundKey_',
 'handleTakeValue_forUnboundKey_',
 'hash',
 'hash',
 'hash',
 'if_setValueIfNonNil_forKey_',
 'if_setValueIfYES_forKey_',
 'ikInMainLoopWait_',
 'imageBrowser_didValidateVisibleCellsAtIndexes_',
 'imageBrowser_willDisplayCellsAtIndexes_',
 'imageSubtitle',
 'imageTitle',
 'imageToDrawForCell_',
 'implementsSelector_',
 'infoForBinding_',
 'init',
 'insertValue_atIndex_inPropertyWithKey_',
 'insertValue_inPropertyWithKey_',
 'int64ValueSafe',
 'int64ValueSafe_',
 'inverseForRelationshipKey_',
 'isCaseInsensitiveLike_',
 'isEqualTo_',
 'isEqual_',
 'isEqual_',
 'isFault',
 'isGreaterThanOrEqualTo_',
 'isGreaterThan_',
 'isKindOfClass_',
 'isLessThanOrEqualTo_',
 'isLessThan_',
 'isLike_',
 'isMemberOfClass_',
 'isNSArray__',
 'isNSCFConstantString__',
 'isNSData__',
 'isNSDate__',
 'isNSDictionary__',
 'isNSNumber__',
 'isNSObject__',
 'isNSOrderedSet__',
 'isNSSet__',
 'isNSString__',
 'isNSTimeZone__',
 'isNSValue__',
 'isNotEqualTo_',
 'isNull',
 'isProxy',
 'isToManyKey_',
 'keyValueBindingForKey_typeMask_',
 'methodDescriptionForSelector_',
 'methodForSelector_',
 'methodSignatureForSelector_',
 'methodSignatureForSelector_',
 'mr_formattedDebugDescription',
 'mutableArrayValueForKeyPath_',
 'mutableArrayValueForKey_',
 'mutableCopy',
 'mutableOrderedSetValueForKeyPath_',
 'mutableOrderedSetValueForKey_',
 'mutableSetValueForKeyPath_',
 'mutableSetValueForKey_',
 'newScriptingObjectOfClass_forValueForKey_withContentsValue_properties_',
 'newTaggedNSStringWithASCIIBytes__length__',
 'objectSpecifier',
 'observationInfo',
 'observeValueForKeyPath_ofObject_change_context_',
 'optionDescriptionsForBinding_',
 'ownsDestinationObjectsForRelationshipKey_',
 'pep_afterDelay_',
 'pep_getInvocation_',
 'pep_onMainThread',
 'pep_onMainThreadIfNecessary',
 'pep_onOperationQueue_',
 'pep_onOperationQueue_priority_',
 'pep_onThread_',
 'pep_onThread_immediateForMatchingThread_',
 'performSelectorInBackground_withObject_',
 'performSelectorOnMainThread_withObject_waitUntilDone_',
 'performSelectorOnMainThread_withObject_waitUntilDone_modes_',
 'performSelector_',
 'performSelector_object_afterDelay_',
 'performSelector_onThread_withObject_waitUntilDone_',
 'performSelector_onThread_withObject_waitUntilDone_modes_',
 'performSelector_withObject_',
 'performSelector_withObject_afterDelay_',
 'performSelector_withObject_afterDelay_ignoreMenuTracking_',
 'performSelector_withObject_afterDelay_inModes_',
 'performSelector_withObject_withObject_',
 'pkaxRespondsToSelector_fromExtrasProtocol_',
 'pkaxValueForKey_',
 'prepareForInterfaceBuilder',
 'pyobjc_performSelectorInBackground_withObject_',
 'pyobjc_performSelectorOnMainThread_withObject_',
 'pyobjc_performSelectorOnMainThread_withObject_modes_',
 'pyobjc_performSelectorOnMainThread_withObject_waitUntilDone_',
 'pyobjc_performSelectorOnMainThread_withObject_waitUntilDone_modes_',
 'pyobjc_performSelector_onThread_withObject_',
 'pyobjc_performSelector_onThread_withObject_modes_',
 'pyobjc_performSelector_onThread_withObject_waitUntilDone_',
 'pyobjc_performSelector_onThread_withObject_waitUntilDone_modes_',
 'pyobjc_performSelector_withObject_afterDelay_',
 'pyobjc_performSelector_withObject_afterDelay_inModes_',
 'receiveObservedError_',
 'receiveObservedValue_',
 'release',
 'release',
 'releaseRenderedTexture_forCGLContext_',
 'removeObject_fromBothSidesOfRelationshipWithKey_',
 'removeObject_fromPropertyWithKey_',
 'removeObservation_',
 'removeObservation_forObservableKeyPath_',
 'removeObserver_forKeyPath_',
 'removeObserver_forKeyPath_context_',
 'removeValueAtIndex_fromPropertyWithKey_',
 'renderToBuffer_withBytesPerRow_pixelFormat_forBounds_',
 'renderWithCGLContext_forBounds_',
 'replaceValueAtIndex_inPropertyWithKey_withValue_',
 'replacementObjectForArchiver_',
 'replacementObjectForCoder_',
 'replacementObjectForKeyedArchiver_',
 'replacementObjectForPortCoder_',
 'resolveForwardingConflictWithPreviousMetadata_forKey_',
 'respondsToSelector_',
 'retain',
 'retain',
 'retainCount',
 'retainCount',
 'retainWeakReference',
 'scriptingProperties',
 'scriptingValueForSpecifier_',
 'self',
 'setAccessibilityBrailleMapRenderRegion_',
 'setAccessibilityBrailleMapRenderer_',
 'setAssociatedObject_',
 'setNilValueForKey_',
 'setObservationInfo_',
 'setObservation_forObservingKeyPath_',
 'setScriptingProperties_',
 'setUserInterfaceItemIdentifier_',
 'setValue_forKeyPath_',
 'setValue_forKey_',
 'setValue_forUndefinedKey_',
 'setValuesForKeysWithDictionary_',
 'settingsView_clearSettingForKey_',
 'settingsView_renameKey_toKey_',
 'settingsView_setSetting_forKey_',
 'settingsView_settingForKey_',
 'settingsView_shouldRenameKey_',
 'shouldColorMatch',
 'storedValueForKey_',
 'stringValueSafe',
 'stringValueSafe_',
 'superclass',
 'supportedBufferPixelFormats',
 'supportedRenderedTexturePixelFormats',
 'supportsBSXPCSecureCoding',
 'supportsRBSXPCSecureCoding',
 'takeStoredValue_forKey_',
 'takeStoredValuesFromDictionary_',
 'takeValue_forKeyPath_',
 'takeValue_forKey_',
 'takeValuesFromDictionary_',
 'toManyRelationshipKeys',
 'toOneRelationshipKeys',
 'un_safeBoolValue',
 'unableToSetNilForKey_',
 'unbind_',
 'userInterfaceItemIdentifier',
 'utf8ValueSafe',
 'utf8ValueSafe_',
 'validateTakeValue_forKeyPath_',
 'validateValue_forKeyPath_error_',
 'validateValue_forKey_',
 'validateValue_forKey_error_',
 'valueAtIndex_inPropertyWithKey_',
 'valueClassForBinding_',
 'valueForKeyPath_',
 'valueForKey_',
 'valueForUndefinedKey_',
 'valueWithName_inPropertyWithKey_',
 'valueWithUniqueID_inPropertyWithKey_',
 'valuesForKeys_',
 'vk_loggingDescription',
 'vk_loggingIdentifier',
 'vk_prettyLoggingDescription',
 'willChangeValueForKey_',
 'willChangeValueForKey_withSetMutation_usingObjects_',
 'willChange_valuesAtIndexes_forKey_',
 'zone']
```

Reply:
Can you please add:

```
# TODO: access directly, e.g. via ApplicationServices.AXUIElementCopyAttributeValue
```